icdiff

```diff
@@ -480,6 +480,16 @@ def create_option_parser():
                       dest='matcher',
                       help="Do not diff lines that match this regex. Not "
                       "compatible with the 'line-numbers' option")
+    parser.add_option("-F", "--filepath-regex",
+                      action="store",
+                      default=None,
+                      type="string",
+                      dest='filepath_regex',
+                      help="Process files (full path) that match this regex pattern")
+    parser.add_option("-I", "--inverse-filepath-regex", default=False,
+                      action="store_true",
+                      help="Inverse regex used in --filepath-regex used to filter"
+                      "files/directories")
```

Comment:
For consistency with `--exclude-lines` which also takes a regex, what would you think of `--exclude-path` and `--include-path`?

The idea would be:

* If `--include-path` is specified, then only paths matching that regexp are included
* If `--exclude-path` is specified, then all paths matching that regexp are not included

I might be confused, but I can't come up with a case where the order of applying `include` and `exclude` matters.

Reply:
Optionally, you could add `--include-lines` to diff only lines matching the regexp, just for even more consistency.